### PR TITLE
ci: add report-missing-fn-doc job, to prevent adding new undocumented pub symbols in vlib

### PR DESCRIPTION
**Pull request by Larpon**
This PR introduce a CI step that'll prevent any *undocumented* `pub fn` functions going into `vlib/`.
It works by comparing the count of undocumented public functions in the fork/branch the PR is forked from with the count of the PR itself. If the count for the PR is greater than that of the fork, CI will fail in the following way:
image

Test runs can be found here from the commmits in this PR's history:

Failed:
https://github.com/Larpon/v/actions/runs/1802505599

Ok (function introduced had docs added):
https://github.com/Larpon/v/runs/5083957234

Currently the numbers speak for themselves:
```
v run cmd/tools/missdoc.v vlib/ | wc -l
1724
```
Currently we have 1724 undocumented functions in vlib - and that number will keep growing if we don't stop it - since very few PR's is currently *adding* documentation.

This PR is, thus, also related to #7047

### CODE REVIEW

File: .github/workflows/docs_ci.yml [hunk 1]
```diff
@@ -16,3 +16,31 @@ jobs:
...
+          ./v run cmd/tools/missdoc.v ./vlib/ > /tmp/n_v.txt
+          ./v run cmd/tools/missdoc.v ./pv/vlib/ > /tmp/o_v.txt
+          count_new=$(cat /tmp/n_v.txt | wc -l)
+          count_old=$(cat /tmp/o_v.txt | wc -l)
+          if [[ ${count_new} -gt ${count_old} ]]; then
+            echo "The following $((count_new-count_old)) function(s) is introduced with no documentation:"
+            sed -n 's/^.*\/vlib\///p' /tmp/n_v.txt > /tmp/tn_v.txt
+            sed -n 's/^.*\/vlib\///p' /tmp/o_v.txt > /tmp/to_v.txt
+            sort /tmp/tn_v.txt /tmp/to_v.txt | uniq -u
+            exit 1
```
Comment (spytheman) [inline .github/workflows/docs_ci.yml:45]: why not simply `diff /tmp/tn_v.txt /tmp/to_v.txt` ?

File: .github/workflows/docs_ci.yml [hunk 2]
```diff
@@ -16,3 +16,31 @@ jobs:
...
+          repository: vlang/v
+          path: pv
+
+      - name: Check against parent commit
+        run: |
+          ./v run cmd/tools/missdoc.v ./vlib/ > /tmp/n_v.txt
+          ./v run cmd/tools/missdoc.v ./pv/vlib/ > /tmp/o_v.txt
+          count_new=$(cat /tmp/n_v.txt | wc -l)
+          count_old=$(cat /tmp/o_v.txt | wc -l)
+          if [[ ${count_new} -gt ${count_old} ]]; then
```
Comment (spytheman) [inline .github/workflows/docs_ci.yml:40]: does that test numerically or alphabetically?

Comment (spytheman) [reply .github/workflows/docs_ci.yml:40]: although for numbers that may not matter actually

Comment (Larpon) [reply .github/workflows/docs_ci.yml:40]: Should be numerically - but bash and `[`/`[[`/`(`/`((` is *tricky* AF

File: .github/workflows/docs_ci.yml (hunk 1 above)
Comment (Larpon) [reply .github/workflows/docs_ci.yml:45]: I couldn't get diff to work locally - I'm testing with this bash script:
```bash
v run ~/Projects/v/cmd/tools/missdoc.v ~/Projects/v/vlib/ > /tmp/n_v.txt
v run ~/Projects/v/cmd/tools/missdoc.v /tmp/v/vlib/ > /tmp/o_v.txt

count_new=$(cat /tmp/n_v.txt | wc -l)
count_old=$(cat /tmp/o_v.txt | wc -l)
if [[ ${count_new} -gt ${count_old} ]]; then
	echo "The following $((count_new-count_old)) function(s) is introduced with no documentation:"
	sed -n 's/^.*\/vlib\///p' /tmp/n_v.txt > /tmp/tn_v.txt
	sed -n 's/^.*\/vlib\///p' /tmp/o_v.txt > /tmp/to_v.txt
	#sort /tmp/tn_v.txt /tmp/to_v.txt | uniq -u
	diff /tmp/tn_v.txt /tmp/to_v.txt
	exit 1
fi
```
And diff outputs a huge list - which I can't figure out why :(

Comment (Larpon) [reply .github/workflows/docs_ci.yml:45]: Ah it's because the list isn't sorted - that'll just introduce an extra `sort` step instead

Comment (Larpon) [reply .github/workflows/docs_ci.yml:45]: I still have problems getting it right even with sort involved

File: .github/workflows/docs_ci.yml (hunk 2 above)
Comment (Larpon) [reply .github/workflows/docs_ci.yml:40]: https://stackoverflow.com/a/42745537/1904615 <- should be numeric `-gt` vs. `<`

File: .github/workflows/docs_ci.yml (hunk 1 above)
Comment (spytheman) [reply .github/workflows/docs_ci.yml:45]: I was just curious - if `uniq -u` works better - so be it

Comment (Larpon) [reply .github/workflows/docs_ci.yml:45]: Ah got it now - hold on

Comment (Larpon) [reply .github/workflows/docs_ci.yml]: Pushed this in b859d9c:

```bash
v run ~/Projects/v/cmd/tools/missdoc.v ~/Projects/v/vlib/ > /tmp/o_v.txt # Local testing
v run ~/Projects/v/cmd/tools/missdoc.v /tmp/v/vlib/ > /tmp/n_v.txt # local testing
#./v run cmd/tools/missdoc.v ./vlib/ > /tmp/n_v.txt
#./v run cmd/tools/missdoc.v ./pv/vlib/ > /tmp/o_v.txt
count_new=$(cat /tmp/n_v.txt | wc -l)
count_old=$(cat /tmp/o_v.txt | wc -l)
if [[ ${count_new} -gt ${count_old} ]]; then
  echo "The following $((count_new-count_old)) function(s) is introduced with no documentation:"
  sed -n 's/^.*\/vlib\///p' /tmp/n_v.txt | sort > /tmp/tn_v.txt
  sed -n 's/^.*\/vlib\///p' /tmp/o_v.txt | sort > /tmp/to_v.txt
  #sort /tmp/tn_v.txt /tmp/to_v.txt | uniq -u
  diff /tmp/tn_v.txt /tmp/to_v.txt
  exit 1
fi
```

Comment (Larpon) [reply .github/workflows/docs_ci.yml]: The output is a little different now (includes a has value???):
```
The following 1 function(s) is introduced with no documentation:
541d540
< os/notify/notify.v:38:0:pub fn undoc() string
```

Comment (Larpon) [reply .github/workflows/docs_ci.yml]: Ah that's just a diff-thing :+1:

**Review by spytheman — APPROVED** [review]
Excellent!

**Pull request merged**